pkg/osquery/tables/xfconf/parser_test.go

```diff
@@ -0,0 +1,262 @@
+							{
+								Name: "LockCommand",
+								Type: "string",
+							},
+						},
+					},
+					{
+						Name: "sessions",
+						Type: "empty",
+						Properties: []Property{
+							{
+								Name: "Failsafe",
+								Type: "empty",
+								Properties: []Property{
+									{
+										Name:  "IsFailsafe",
+										Type:  "bool",
+										Value: "true",
+									},
+									{
+										Name:  "Count",
+										Type:  "int",
+										Value: "5",
+									},
+									{
+										Name: "Client0_Command",
+										Type: "array",
+										Values: []ArrayValue{
+											{
+												Type:  "string",
+												Value: "xfwm4",
+											},
+										},
+									},
+									{
+										Name:  "Client0_Priority",
+										Type:  "int",
+										Value: "15",
+									},
+									{
+										Name:  "Client0_PerScreen",
+										Type:  "bool",
+										Value: "false",
+									},
+									{
+										Name: "Client1_Command",
+										Type: "array",
+										Values: []ArrayValue{
+											{
+												Type:  "string",
+												Value: "xfsettingsd",
+											},
+										},
+									},
+									{
+										Name:  "Client1_Priority",
+										Type:  "int",
+										Value: "20",
+									},
+									{
+										Name:  "Client1_PerScreen",
+										Type:  "bool",
+										Value: "false",
+									},
+									{
+										Name: "Client2_Command",
+										Type: "array",
+										Values: []ArrayValue{
+											{
+												Type:  "string",
+												Value: "xfce4-panel",
+											},
+										},
+									},
+									{
+										Name:  "Client2_Priority",
+										Type:  "int",
+										Value: "25",
+									},
+									{
+										Name:  "Client2_PerScreen",
+										Type:  "bool",
+										Value: "false",
+									},
+									{
+										Name: "Client3_Command",
+										Type: "array",
+										Values: []ArrayValue{
+											{
+												Type:  "string",
+												Value: "Thunar",
+											},
+											{
+												Type:  "string",
+												Value: "--daemon",
+											},
+										},
+									},
+									{
+										Name:  "Client3_Priority",
+										Type:  "int",
+										Value: "30",
+									},
+									{
+										Name:  "Client3_PerScreen",
+										Type:  "bool",
+										Value: "false",
+									},
+									{
+										Name: "Client4_Command",
+										Type: "array",
+										Values: []ArrayValue{
+											{
+												Type:  "string",
+												Value: "xfdesktop",
+											},
+										},
+									},
+									{
+										Name:  "Client4_Priority",
+										Type:  "int",
+										Value: "35",
+									},
+									{
+										Name:  "Client4_PerScreen",
+										Type:  "bool",
+										Value: "false",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range testCases {
+		result, err := readChannelXml(tt.filePath)
```

Comment:
```suggestion
	for _, tt := range testCases {
	        tt := tt
		result, err := readChannelXml(tt.filePath)
```

Pretty sure we need the tt thing.

Consider, though not required, `t.Run(...)`

Reply:
I agree with using `t.Run(...)`. If I understand corrected the `tt := tt` is used to copy the test case before a call to `t.Parallel()` inside `t.Run(...)`. Without out the `tt := tt`, the original `tt` declared in the range statement has the potential to get replaced before a previous `t.Run(...)` has completed.

Reply:
_Ah_. Yes, James is right -- we only need `tt := tt` if there's a goroutine (or t.Run) in the mix. Otherwise it's fine.

(Side note, I think this is changing in a future go version)

Reply:
Updated to do this + `t.Run(...)` -- thank you for the explanation 🙂